ci-automation/packages.sh

```diff
@@ -80,6 +80,18 @@ function _packages_build_impl() {
     local vernum="${FLATCAR_VERSION}"
     local docker_vernum="$(vernum_to_docker_image_version "${vernum}")"
     local packages_container="flatcar-packages-${arch}-${docker_vernum}"
+    local torcx_pkg_url="https://${BUILDCACHE_SERVER}/images/${arch}/${vernum}/torcx"
+
+    source sdk_lib/sdk_container_common.sh
+
+    if is_official "${vernum}"; then
+        # A channel returned by get_git_channel should not ever be
+        # "developer" here, because it's an official build done from
+        # one of the maintenance branches. So if the channel happens
+        # to be "developer", then you are doing it wrong (releasing
+        # from the main branch?).
+        torcx_pkg_url="https://$(get_git_channel).release.flatcar-linux.net/${arch}-usr/${vernum}/torcx"
```

Comment:
Does this need to be available during the kola tests already?

Reply:
Hm, well spotted. At first I thought that the test does a shallow check of the file structure, but it actually tries to download the packages with ignition and runs some basic docker tests on those.

So this won't fly - old pipeline was doing the uploading while right after building the torcx store, and it uploaded right away at to the final location. New pipeline uploads the packages to the staging area (bincache) and for releases my changes would expect the packages to be already on the origin server. I'll sleep with it.

Reply:
The torcx package URL needs to stay like I wrote it, because for releases, the packages will eventually be on origin, not in bincache.  And while writing it I noticed that we have [a `__prepare_torcx` function](https://github.com/flatcar-linux/scripts/blob/6d8d88cbe4db9a998dcacfdeb831d708fe525292/ci-automation/test.sh#L77-L100) that already mangles the downloaded torcx manifest by adding the URLs to the bincache, because otherwise the test would be failing. Interesting.

So I plan to change this mangling a bit to replace "release" URLs with "bincache" URLs.